cmd/bucket-notification-handlers.go

```diff
@@ -248,7 +249,7 @@ func (api objectAPIHandlers) ListenBucketNotificationHandler(w http.ResponseWrit
 
 	<-target.DoneCh
 
-	if err := RemoveListener(objAPI, bucketName, target.ID(), *thisAddr); err != nil {
+	if err := RemoveListener(ctx, objAPI, bucketName, target.ID(), *thisAddr); err != nil {
```

Comment:
1. If same context can be used for two calls, we don't need to create new context everywhere because of `ctx` is created top of every handler methods.  If it is not correct to use the same context, this needs to be fixed.

2. You would need to use object API methods for Save/Remove Listeners and CRUD methods of notification configuration everywhere.

Reply:
fixed this.